mlflow/utils/credentials.py

```diff
@@ -183,5 +191,18 @@ def _databricks_login():
     profile_name = os.environ.get("DATABRICKS_CONFIG_PROFILE", "DEFAULT")
     _overwrite_or_create_databricks_profile(file_name, profile, profile_name)
 
-    if not _check_databricks_auth():
-        raise MlflowException("Failed to sign in Databricks, please retry `mlflow.login()`.")
+    try:
+        # Failed validation will throw an error.
+        _validate_databricks_auth()
+        # Connect to Databricks if the credentials are valid.
+        set_tracking_uri("databricks")
```

Comment:
Do we execpt this line to throw? If not, I'd pull this out from try-catch.

Reply:
For example, the current code throws `Failed to sign in Databricks` if `set_tracking_uri` fails, but that's not correct.

Reply:
Yea I thought about it, basically it only throws error when the credentials are invalid.

Reply:
It actually doesn't even throw. It sets the tracking URI.

Reply:
you are right, `set_experiment` throws error instead of `set_tracking_uri`

Reply:
but there is no guarantee that `set_tracking_uri` will stay as it is. I might add code that can throw tomorrow.

Reply:
hmmm every API is subject to change.. we may want to stick to the status quo. If the API changes we can make changes accordingly.

Reply:
but yea, I am down to moving it out of the try block.

Reply:
So we need to update this function when we update `set_tracking_uri`?